Impossible-Agent-Suite/SwapFees/src/pairs.fetcher.ts

```diff
@@ -0,0 +1,27 @@
+import { providers, Contract, BigNumber } from "ethers";
+import abi from "./abi";
+
+export default class PairFetcher {
+  readonly factory: string;
+  private fContract: Contract;
+
+  constructor(factory: string, provider: providers.Provider) {
+    this.factory = factory;
+    this.fContract = new Contract(factory, abi.FACTORY, provider);
+  }
+
+  public async getAllPairs(block: number | string): Promise<string[]> {
```

Comment:
Is there any point in specifying the particular block if you'll only ever be accessing the latest block in `agent.ts`?

Reply:
That's true, but I'll keep it that way. This is the general signature that I always use for functions that do some blockchain interaction.